docs/specification.md

```diff
@@ -1,150 +1,367 @@
+  1. Flexibility and backwards compatibility of sequence attributes. What happens for an implementation that adds a new attribute? For example, if an implementation adds a `topology` attribute to the sequences, in the sequence-oriented structure, this would alter the sequence object and thereby change its digest. In the array-based structure, since we digest the arrays individually, the array digest is not changed. Thus, the array-oriented structure emphasizes flexibility of attributes, where the sequence-oriented structure would emphasize flexibility of sequences. In other words, the array-based structure makes it straightforward to mix-and-match *attributes* of the collection. Because each attribute is independent, and not integrated into individual sequence objects, it is simpler to select and build subsets and permutations of attributes. We reasoned that flexibility of attributes was desirable.
 
-- `ATTRIBUTE_DELIMITER` is defined as: `undecided`
-- `ITEM_DELIMITER` is defined as: `undecided`
-- `hash function` is allowed to be either `MD5` OR `TRUNC-512`. Details to follow.
+   2. Conciseness. Sequence collections may be used for tens of thousands or even millions of sequences. For example, a transcriptome may contain a million transcripts. The array-oriented data structure is a more concise representation for digesting collections with many elements because the attribute names are specified once *per collection* instead of once *per element*. Furthermore, the level 1 representation of the sequence collection is more concise for large collections, since we only need one digest *per attribute*, rather than one digest *per sequence*. 
 
-### 2. Retrieval: Retrieving sequence collections from digests
+   3. Utility of intermediate digests. The array-oriented approach provides useful intermediate digests for each attribute. This digest can be used to test for matching sets of sequences, or matching coordinate systems, using the individual component digests. With a sequence-oriented framework, this would require traversing down a layer deeper, to the individual elements, to establish identity of individual components. The alternative advantage we would have from a sequence-oriented structure would be identifiers for *annotated sequences*. We gain the advantages of these digests through the *names-lengths* attribute.
 
-The retrieval function is fulfilled by an API specification that provides an ability to retrieve original sequences from a database keyed by the unique digest.
 
-#### Core endpoints
+### F2. Collated attributes
 
-Core endpoints are `REQUIRED` for an implementation to be compliant with the seqcol protocol.
+In jsonschema, there are 2 ways to qualify properties: 1) a local qualifier, using a key under a property; or 2) an object-level qualifier, which is specified with a keyed list of properties up one level. For example, you annotate a property's `type` with a local qualifier, underneath the property, like this:
```

Comment:
"json schema" is two words (at least according to https://json-schema.org/
(in fact, possibly it should be "JSON schema"?)